homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       CASE WHEN salary >= (SELECT AVG(salary)
+                            FROM employees) THEN 'HIGH'
+            ELSE 'LOW' END AS SalaryStatus
+FROM employees;
+
 --33. the employee id, name ( first name and last name ), SalaryDrawn, AvgCompare (salary - the average salary of all employees)
     -- and the SalaryStatus column with a title HIGH and LOW respectively for those employees whose salary is more than and less than
     -- the average salary of all employees.
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name,
+       salary * commission_pct AS SalaryDrawn,
+       CASE WHEN salary >= avg_salary THEN 'HIGH'
+          ELSE 'LOW' END AS SalaryStatus,
+       salary - avg_salary as AvgCompare
+FROM employees,
+    (
+    SELECT AVG(salary) AS avg_salary
+    FROM employees
+);
+
 --34. all the employees who earn more than the average and who work in any of the IT departments.
+SELECT *
+FROM employees
+WHERE salary > (
+    SELECT AVG (salary)
+    FROM employees
+)
+  AND department_id IN (
+    SELECT department_id
+    FROM departments
+    WHERE department_name LIKE '%IT%'
+);
+
 --35. who earns more than Mr. Ozer.
+SELECT *
+FROM employees
+WHERE salary > (
+    SELECT salary
+    FROM employees
+    WHERE last_name = 'Ozer'
+    );
+
 --36. which employees have a manager who works for a department based in the US.
+SELECT *
+FROM employees
+WHERE manager_id IN (
+    SELECT manager_id
+    FROM departments
+    WHERE location_id IN (
+        SELECT location_id
+        FROM locations
+        WHERE country_id  = 'US'
+    ) AND manager_id IS NOT NULL
+);
+
 --37. the names of all employees whose salary is greater than 50% of their department’s total salary bill.
+SELECT (first_name || ' ' || last_name) AS employee_name
+FROM employees
+         INNER JOIN (
+             SELECT department_id, SUM(salary) AS total_salary
+             FROM employees
+             GROUP BY department_id
+) USING (department_id)
+WHERE salary > total_salary * 0.5;
+
 --38. the employee id, name ( first name and last name ), salary, department name and city for all
---the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.  
+SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, salary, department_name, city
+FROM employees
+    INNER JOIN departments USING (department_id)
+    INNER JOIN locations USING (location_id);
+
+--the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.
+SELECT *
+FROM employees
+WHERE salary = (
+    SELECT MAX(salary)
+    FROM employees
+    WHERE hire_date BETWEEN DATE '2002-01-01' AND DATE '2003-12-31'
+    );
```

Comment:
I guess it should look like this:
```suggestion
--the employees who gets the salary as the salary earn by the employee which is maximum within the joining person January 1st, 2002 and December 31st, 2003.
SELECT employee_id, (first_name || ' ' || last_name) AS employee_name, salary, department_name, city
FROM employees
    INNER JOIN departments USING (department_id)
    INNER JOIN locations USING (location_id)
WHERE salary = (
    SELECT MAX(salary)
    FROM employees
    WHERE hire_date BETWEEN DATE '2002-01-01' AND DATE '2003-12-31'
    );
```

Reply:
Done